Sources/Networking/v2/Extensions/Dictionary+URLQueryItem.swift

```diff
@@ -0,0 +1,35 @@
+//
+//  Dictionary+URLQueryItem.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+
+extension Dictionary where Key == String, Value == String {
```

Comment:
can we make it a `Collection` extension defaulting to `KeyValuePairs<String, String>` – this will preserve items order

Reply:
Good suggestion thanks, I agree that maintaining the order is a nice to have, I tried to change it in 
`public typealias QueryItems = any Collection<KeyValuePairs<String, String>>`
But that introduces a lot of issues (like can't access values directly by key), then I decided to stuck with

```
public typealias QueryItem = Dictionary<String, String>.Element
public typealias QueryItems = Array<QueryItem>
```
Less elegant but achieves the objective of having a data structure of ordered key value pairs

I moved everything to `QueryItems.swift`